dig/dig.go

```diff
@@ -62,15 +89,23 @@ func (g *Graph) Register(c interface{}) error {
 
 	switch ctype.Kind() {
 	case reflect.Func:
-		if ctype.NumOut() != 1 {
+		if ctype.NumOut() == 0 || ctype.NumOut() > 2 {
```

Comment:
Yeah, I updated this. It is now,

> The provided argument must be a function that accepts its dependencies as
arguments and returns a single result, which must be a pointer type.
The function may optionally return an error as a second result.